### ISSUE TITLE
Some Decky Plugin NSL shortcuts not launching

### ISSUE THREAD

**Issue by Shadow2lead** [moraroy/NonSteamLaunchers-On-Steam-Deck#416]
I used the decky plugin to install games from gog. Some are working: click the shortcut in Non-Steam in game mode, gog launches, the game auto launches. For others I click the shortcut, click play and the screen flashes and just returns to the steam game screen without launching the game. However, if I run gog and run the game from within gog then the game runs. 

Problem games: 
Heroes of Might and Magic
Possibly Heroes of Might and Magic II (I think it created two shortcuts to heroes 1, and I don't actually have a shortcut to 2)
 Star Wars Episode 1 Racer

I have tried restarting the steam deck, but the links in steam still did not work for those games (they can still run directly from gog Galaxy)

**Comment by moraroy** [comment]
@Shadow2lead yeah it may be because it isnt parsing those games correctly for whatever reason, there is no dlc involved correct?

**Comment by Shadow2lead** [comment]
Correct, no dlc involved. All 90s or early 00s games. I'll have to double check later, but I think a few of the games with the problems run via dosbox, but at least one (Racer) I'm almost certain doesn't.

**Comment by moraroy** [comment]
@Shadow2lead oh dang im not sure if i even have dos box games on my gog... how can i check that in the gog launcher?

**Comment by Shadow2lead** [comment]
If you go to manage installation, show folder, there is a "DOSBOX" folder on each I've checked, but also when you start the game you'd see a dosbox window pop up briefly as that starts up to then run the game.

**Comment by notimp** [comment]
If you know your way around dos commands, and want to tinker, you can always make dosbox game launch in the native discoverstore dosbox.

https://old.reddit.com/r/SteamDeck/comments/12v0kog/all_tex_murphy_games_played_on_steam_deck/

Dosbox repackagings in gog sometimes are very creative. As in launch the windows dosxbox exe with two configuration files while the config files are pointing to other files, which are dependent on being in a fixed folder location and other fun stuff like that.

**Comment by moraroy** [comment]
@notimp @Shadow2lead to be honest I figured out why they aren't launching at least on my end, there is a small syntaxes error, I recently got my first dos box game for gog this past week. Been doing a lot of testing. And I found an error in my syntax.

**Comment by moraroy** [comment]
@notimp but you are correct in the commands this is very helpful thank you <3

**Comment by moraroy** [comment]
@Shadow2lead @notimp allright so.... i just pushed a bunch of really cool updates, still not done yet, BUT... good news, currently right now, all DOS Box games are correctly formatted now for the desktop and the plugin version of NSL! i would grab the latest version and make sure you delete your old shortcuts and lets NSL make the new ones. The dos box games seem to "run" in game mode no problem. but currently though, in desktop mode, i believe if you were to try to launch the dos box games, it may be "blank" or have a black screen, this is very easy to fix, you simply go into the games ".conf" file. and set Fullscreen to True.

by default GOG seems to be installing the games with a default .conf on the deck which seems to be the 1980x960.conf located in the gog galaxy games folder. this means that is the one you want to edit for your dos box game settings. as @notimp described there is further settings to adjust for each game, but im trying to make it as easy as possible for the steam deck users given the use of the gog launcher. let me know how it works out
